qiskit_ibm_runtime/estimator.py

```diff
@@ -104,16 +107,17 @@ class EstimatorV2(BasePrimitiveV2[EstimatorOptions], Estimator, BaseEstimatorV2)
 
     def __init__(
         self,
-        backend: Optional[Union[str, IBMBackend]] = None,
+        mode: Optional[Union[IBMBackend, Session, Batch]] = None,
+        backend: Optional[IBMBackend] = None,
```

Comment:
This is going to be a bit messy. For one, we can't just drop the support of `str` as a valid type for `backend`. And if you are going to make `mode` the new first positional argument, it needs to accept types the old first arg (i.e. `backend`) supports. So existing code like `Estimator("ibm_foo")` won't break. But you can still issue a deprecation message in this case. 

```suggestion
        mode: Optional[Union[IBMBackend, Session, Batch, str]] = None,
        backend: Optional[str, IBMBackend] = None,
```